src/Catalyst.Core.Modules.Web3/Controllers/LedgerController.cs

```diff
@@ -39,37 +42,46 @@ public sealed class LedgerController : Controller
     {
         private readonly IDeltaHashProvider _deltaHashProvider;
         private readonly IDfs _dfs;
+        private readonly IMapperProvider _mapperProvider;
         private readonly ILogger _logger;
 
-        public LedgerController(IDeltaHashProvider deltaHashProvider, IDfs dfs, ILogger logger)
+        public LedgerController(IDeltaHashProvider deltaHashProvider, IDfs dfs, IMapperProvider mapperProvider, ILogger logger)
         {
             _deltaHashProvider = deltaHashProvider;
             _dfs = dfs;
+            _mapperProvider = mapperProvider;
             _logger = logger;
         }
 
         [HttpGet]
         public async Task<JsonResult> GetLatestDelta(DateTime? asOf)
         {
             var latest = _deltaHashProvider.GetLatestDeltaHash(asOf?.ToUniversalTime());
-            byte[] dfsContent = null;
             try
             {
                 using (var fullContentStream = await _dfs.ReadAsync(latest))
                 {
-                    dfsContent = await fullContentStream.ReadAllBytesAsync(CancellationToken.None);
+                    var contentBytes = await fullContentStream.ReadAllBytesAsync(CancellationToken.None);
+                    var delta = Delta.Parser.ParseFrom(contentBytes).ToDao<Delta, DeltaDao>(_mapperProvider);
+
+                    return Json(new
+                    {
+                        Success = true,
+                        DeltaHash = latest,
+                        Delta = delta
+                    });
                 }
             }
             catch (Exception e)
             {
-                _logger.Error(e, "Failed to find dfs content for delta as of {asOf} at {dfsTarget}", asOf, latest);
+                var errorMessage = $"Failed to find dfs content for delta as of {asOf} at {latest}";
```

Comment:
you lost the structured logging by passing a raw string here... no big deal, but worth keeping in mind (cf. https://serilog.net/)

Reply:
Cheers, I thought it was just a simple format wrapper, didn't realise it provided a better format structure. Updated based on your feedback.